src/attached_probe.cpp

```diff
@@ -753,7 +753,9 @@ void AttachedProbe::load_prog(BPFfeature &feature)
         std::string btf_name = name;
         if (probe_.type == ProbeType::iter)
           btf_name = "bpf_iter_" + btf_name;
-        opts.attach_btf_id = btf_.get_btf_id(btf_name);
+        auto btf_id = btf_.get_btf_id_fd(btf_name);
+        opts.attach_btf_id = btf_id.first;
+        opts.attach_prog_fd = btf_id.second;
```

Comment:
Can we use the other union member, `attach_btf_obj_fd` instead? In the libbpf headers, they're actually two separate members.

Further, I actually think we should explicitly convert the ID to an FD here, since this is the only place where we need it. Right now, the code is acquiring FDs for everything and not using them in 99% of cases.

Reply:
> Can we use the other union member, `attach_btf_obj_fd` instead? In the libbpf headers, they're actually two separate members.

Well, it won't work for libbpf<0.6 which we still support. But I'm fine dropping that support since we plan to move to higher libbpf anyway.

> Further, I actually think we should explicitly convert the ID to an FD here, since this is the only place where we need it. Right now, the code is acquiring FDs for everything and not using them in 99% of cases.

In fact we do, it's just hidden inside `get_btf_id_fd` which is only called from here (and from `bpffeature` when trying to load some programs). But I agree that it may be cleaner to do the conversion in this function, especially if the function will be closing it (see my question about closing the FD below).